pkg/util/archive/zip_test.go

```diff
@@ -0,0 +1,179 @@
+	assert.Nil(t, err)
+	err = os.WriteFile(filepath.Join(nestedDirectory, "nested_file.txt"), []byte("Hello\n"), 0755)
+	assert.Nil(t, err)
+
+	err = Zip([]string{target, symlink, nestedDirectory}, zipTempLocation)
+
+	assert.Nil(t, err)
+
+	// Unzip the archive
+	destinmationDir := t.TempDir()
+	err = UnZip(zipTempLocation, destinmationDir)
+	assert.Nil(t, err)
+
+	// 'symtarget.txt' safely extracted without errors inside the destination path
+	_, err = os.Stat(filepath.Join(destinmationDir, "symtarget.txt"))
+	assert.Nil(t, err, "symtarget.txt should be extracted inside the destination folder")
+
+	// 'symlink' is not zipped
+	_, err = os.Stat(filepath.Join(destinmationDir, "symlink"))
+	assert.True(t, os.IsNotExist(err))
+
+	// 'nested/nested_file.txt' is not a symlink and should be extracted without errors inside the destination path
+	_, err = os.Stat(filepath.Join(tmpDir, "nested", "nested_file.txt"))
+	assert.Nil(t, err, "nested/nested_file.txt should be extracted inside the destination folder")
+}
+
+func TestUnZip(t *testing.T) {
+	destinationZip := createUnSafedZip(t, false)
+
+	tmpDir := t.TempDir()
+
+	err := UnZip(destinationZip, tmpDir)
+
+	assert.Nil(t, err)
+
+	// 'goodfile.txt' safely extracted without errors inside the destination path
+	_, err = os.Stat(filepath.Join(tmpDir, "goodfile.txt"))
+	assert.Nil(t, err, "goodfile.txt should be extracted inside the destination folder")
+
+	// 'bad/file.txt' is not a symlink and should be extracted without errors inside the destination path
+	fileInfo, err := os.Stat(filepath.Join(tmpDir, "bad", "file.txt"))
+	assert.Nil(t, err, "bad/file.txt should be extracted inside the destination folder")
+	assert.True(t, fileInfo.Mode() != os.ModeSymlink)
+}
+
+func TestUnZip_OutsideRoot(t *testing.T) {
+	destinationZip := createUnSafedZip(t, true)
+
+	err := UnZip(destinationZip, "fixtures")
+
+	assert.Error(t, err, "illegal file path: ../badfile.txt")
+}
+
+type file struct {
+	Name, Body string
+}
+
+func createUnSafedZip(t *testing.T, createFileOutsideRoot bool) string {
+	// Create a buffer to write our archive to.
+	tmpDir := t.TempDir()
+
+	path := filepath.Join(tmpDir, "evil.zip")
+	fw, err := os.Create(path)
+	if nil != err {
+		t.Fatalf("Failed to create zip file: %s", err)
+	}
+
+	// Create a new zip archive.
+	w := zip.NewWriter(fw)
+
+	// Write the evil symlink
+	h := &zip.FileHeader{
+		Name:     "bad/file.txt",
+		Method:   zip.Deflate,
+		Modified: time.Now(),
+	}
+	h.SetMode(os.ModeSymlink)
+	header, err := w.CreateHeader(h)
+	if err != nil {
+		t.Fatalf("Failed to create file header: %s", err)
+	}
+	// The evil symlink points outside of the target directory
+	_, err = header.Write([]byte("../../badfile.txt"))
+	if err != nil {
+		t.Fatalf("Failed to write file: %s", err)
+	}
+
+	// Write safe files to the archive.
+	var files = []file{
+		{"goodfile.txt", "hello world"},
+		{"morefile.txt", "hello world"},
+		{"bad/file.txt", "Mwa-ha-ha"},
+	}
+
+	if createFileOutsideRoot {
```

Comment:
We should also try at least minimally testing absolute paths (e.g. `/usr/bin/badfile.sh`)